src/ui/buttons/script.jsx

```diff
@@ -166,11 +163,21 @@ function getComponentScript() : () => void {
 }
 
 type ScriptProps = {|
-    nonce : ?string
+    nonce : ?string,
+    // eslint-disable-next-line flowtype/no-weak-types
+    buttonAnimation : any
 |};
 
-export function Script({ nonce } : ScriptProps) : ElementNode {
+export function Script({ nonce, buttonAnimation } : ScriptProps) : ElementNode {
+    const htmlForScript = `
```

Comment:
Can you pull some of this js logic out before defining the htmlForScript? Or add comments explaining what i going on because it is not very readable